compiler/lib/flow.ml

```diff
@@ -337,6 +337,28 @@ let the_def_of info x =
         x
   | Pc c -> Some (Constant c)
 
+(* If [constant_identical a b = true], then the two values cannot be
+   distinguished, i.e., they are not different objects (and [caml_js_equals a b
+   = true]) and if both are floats, they are bitwise equal. *)
+let constant_identical a b =
+  match a, b with
+  | Int i, Int j -> Int32.equal i j
+  | Float a, Float b -> Float.bitwise_equal a b
+  | NativeString a, NativeString b -> Native_string.equal a b
+  | String a, String b -> Config.Flag.use_js_string () && String.equal a b
```

Comment:
I expect that we may have to update this for Wasm.

Reply:
Indeed, this should be false for wasm_of_ocaml.
Maybe we can just drop this case?

Reply:
Which one?

Reply:
Native strings are JavaScript strings. We have a hack to use JavaScript equality for them, so the first line is fine.

Also, since floats are boxed, one can distinguish two floats with the same value.

Reply:
Maybe this suggests that the function should behave differently between the two backends (once there are two backends)?